### PR TITLE
Fix UNICODE error

### PR DESCRIPTION
**Pull request by konstest**
### What does this PR do?
Fixed error if VMware vCenter is a virtual machine with the Unicode characters

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/44452

### Tests written?
Сhecked by hand

### Commits signed with GPG?
No

### CODE REVIEW

File: salt/cloud/__init__.py [hunk 1]
```diff
@@ -1494,7 +1494,7 @@ def do_action(self, names, kwargs):
                         else:
                             log.debug(
                                 'vm:{0} in provider:{1} is not in name '
-                                'list:\'{2}\''.format(vm_name, driver, names)
+                                'list:\'{2}\''.format(vm_name.encode('utf-8'), driver, names)
```
Comment (cachedout) [inline salt/cloud/__init__.py]: This should likely be `__salt_system_encoding__` instead of `utf-8`. That will select the correct encoding type.

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/cloud/__init__.py (hunk 1 above)
Comment (damon-atkins) [reply salt/cloud/__init__.py]: This is the wrong fix.
from __future__ import unicode_literals
   OR
prefix all strings with u'string'

The reason this is broken is discussed in this large issue https://github.com/saltstack/salt/issues/42278#issuecomment-318913952

The long term fix is to ensure everything is in Unicode Format i.e. salt.cloud.CloudClient always returns Unicode no matter what the driver is and takes Unicode and the drive converts down from Unicode if it needs to.

The fix I suggest allows logging to work with mixed PY2 and PY2 Unicode.

Comment (rallytime) [reply salt/cloud/__init__.py]: @konstest Can you change this to `__salt_system_encoding__` as @cachedout suggested above? The other comments from @damon-atkins are being handled in a different way right now.

Once that is updated, we can get this fix in.

**Comment by rallytime** [comment]
bump @konstest

**Comment by konstest** [comment]
@rallytime 
Fixed

File: salt/cloud/__init__.py [hunk 2]
```diff
@@ -1500,7 +1500,7 @@ def do_action(self, names, kwargs):
                         else:
                             log.debug(
                                 'vm:{0} in provider:{1} is not in name '
-                                'list:\'{2}\''.format(vm_name, driver, names)
+                                'list:\'{2}\''.format(vm_name.encode(__salt_system_encoding__), driver, names)
```
Comment (damon-atkins) [inline salt/cloud/__init__.py]: Why would we want to convert Unicode to bytes and then have log output bytes?

**Review by damon-atkins — CHANGES_REQUESTED** [review]

**Comment by damon-atkins** [comment]
The correct fix is `export LANG=en_US.UTF-8` we should trap this and suggest using a UTF-8 Language setting.

**Comment by rallytime** [comment]
My apologies for not coming back to this sooner! I think the fix in #46455 is probably the better way to go here. Thanks for submitting this fix @konstest, but I am going to close this in favor of #46455.

**Pull request closed without merging**